telemetry-core/src/main/java/com/newrelic/telemetry/events/EventBatchSender.java

```diff
@@ -79,7 +81,19 @@ public static EventBatchSender create(SenderConfiguration configuration) {
     Utils.verifyNonNull(configuration.getApiKey(), "API key cannot be null");
     Utils.verifyNonNull(configuration.getHttpPoster(), "an HttpPoster implementation is required.");
 
-    URL url = configuration.getEndpointUrl();
+    String userRegion = configuration.getRegion();
+    boolean isCustomEndpoint = configuration.isUserProvideEndpoint();
+
+    URL url = null;
+    if (isCustomEndpoint == true) {
```

Comment:
please add a test in each BatchSender to confirm custom endpoint will override region